heed/src/env.rs

```diff
@@ -180,22 +184,29 @@ impl EnvOpenOptions {
     pub fn open<P: AsRef<Path>>(&self, path: P) -> Result<Env> {
         let mut lock = OPENED_ENV.write().unwrap();
 
-        let path = match canonicalize_path(path.as_ref()) {
+        let (path, handle) = match canonicalize_path(path.as_ref()) {
             Err(err) => {
                 if err.kind() == NotFound && self.flags & (Flag::NoSubDir as u32) != 0 {
                     let path = path.as_ref();
                     match path.parent().zip(path.file_name()) {
-                        Some((dir, file_name)) => canonicalize_path(dir)?.join(file_name),
+                        Some((dir, file_name)) => {
+                            let handle = Handle::from_path(dir)?;
+                            let path = canonicalize_path(dir)?.join(file_name);
+                            (path, handle)
+                        }
                         None => return Err(err.into()),
                     }
                 } else {
                     return Err(err.into());
                 }
             }
-            Ok(path) => path,
+            Ok(path) => {
+                let handle = Handle::from_path(&path)?;
+                (path, handle)
+            }
         };
```

Comment:
@Kerollmops I am wondering if we still need canonisation, for `Flag::NoSubDir` it's mandatory I think, but for the other possibilities?

Reply:
We needed to go to the parent, to be frank I didn't thought this part preferred to open the discussion! :)

Reply:
I am pretty sure that you can entirely remove the `canonicalize_path` function even when we use the `NoSubDir` flag. Could you create an issue to track this, please? We can prioritize this later (and specify that we must add strange tests about strange paths).

Reply:
https://github.com/meilisearch/heed/issues/180 don't hesitate to edit the issue and it's own scope.